### ISSUE TITLE
npm --init fails whenever trying to create a package with a name starting with a capital letter

### ISSUE THREAD

**Issue by ccheever** [npm/npm#8120]
When `npm init` is run, the first question that the interactive form asks is for the name of the package.
If a user enters a package name that starts with a capital letter, the script blows up with a cryptic error.

Here is an example:
```
ccheever@Charless-Air:~/tmp/StartsWithACapitalLetter$npm init
This utility will walk you through creating a package.json file.
It only covers the most common items, and tries to guess sensible defaults.

See `npm help json` for definitive documentation on these fields
and exactly what they do.

Use `npm install <pkg> --save` afterwards to install a package and
save it as a dependency in the package.json file.

Press ^C at any time to quit.
name: (StartsWithACapitalLetter)
npm ERR! Darwin 14.3.0
npm ERR! argv "/Users/ccheever/.nvm/versions/io.js/v1.8.1/bin/iojs" "/Users/ccheever/.nvm/versions/io.js/v1.8.1/bin/npm" "init"
npm ERR! node v1.8.1
npm ERR! npm  v2.8.4

npm ERR! Cannot read property 'join' of undefined
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/ccheever/tmp/StartsWithACapitalLetter/npm-debug.log
```

I believe the script should either:
(1) allow the creation of packages that start with capital letters (there doesn't seem to be any obvious problem with this, though it certainly isn't the convention, at least for things being published to npm)
or 
(2) Give a comprehensible and actionable error message to the user (i.e. not `Cannot read property 'join' of undefined`)

I'm not sure which of these is right; if someone tells me, I can try to come up with a patch and submit a PR.

This reproduces all the time on iojs v1.8.1 using npm v2.8.4 (though I haven't tried other versions of iojs/node yet).

**Comment by othiym23** [comment]
This is fixed in `npm@2.9.0` (which is currently in testing – run `npm -g install npm@next` if you want to verify the fix for yourself). There are two things going on:

1. `npm init` was not handling errors returned by package name validation correctly in all cases. See npm/init-package-json@e062308d11df589b700289ce106c1e74fed43875 for the fix, which was landed in npm in faf65a7bbb2fad13216f64ed8f1243bafe743f97.
2. new package names cannot contain capital letters, which is documented in a few places, but probably needs to be made clearer and more prominent. Some legacy packages have caps in their names (which the npm CLI needs to be fixed to support again), but no new packages can have caps in their names. There will probably be a tweak to the npm documentation soon to make this clearer.

**Comment by ccheever** [comment]
Got it; thanks.
